conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,481 @@
+Description: Test let-expr with query expressions.
+Labels: let-expr, record-type, array-type, list-constructor-expr, module-type-decl,
+
+type Department record {|
+   int id;
+   string name;
+|};
+
+type Person record {|
+   int id;
+   string fname;
+   string lname;
+|};
+
+type DeptPerson record {|
+   string fname;
+   string lname;
+   string? dept;
+|};
+
+Person[] personList = [
+    {id: 1, fname: "Alex", lname: "George"}
+];
+
+Department[] deptList = [
+    {id: 1, name:"HR"}
+];
+
+function init() {
+    var l = let Person[] personList = [{id: 1, fname: "Sam", lname: "George"}],
+                   Department[] deptList = [{id: 1, name:"HR"}] in
+                   from var person in personList
+                        join var {id: deptId, name: deptName} in deptList
+                        on person.id equals deptId
+                        select {
+                            fname : person.fname,
+                            lname : person.lname,
+                            dept : deptName
+                        };
+    io:println(l); // @output [{"fname":"Sam","lname":"George","dept":"HR"}]
+}
+
+Test-Case: parser-error
+Description: let-expr with missing tokens
+Labels: let-expr
+
+function errorFunction() {
+    io:println(let int i in i); // @error missing equal token
+                                // @error missing identifier
+    io:println(let i = 2 in i); // @error unknown type 'i'
+                                // @error missing identifier
+                                // @error undefined symbol 'i'
+    io:println(let int i = 2 i); // @error missing in keyword
+    io:println(let int i = 2 int j = 2 in i + j); // @error missing comma token
+    io:println(let in i); // @error missing let variable declaration
+                          // @error undefined symbol 'i'
+    io:println(let i); // @error unknown type 'i'
+                       // @error missing equal token
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing identifier
+                       // @error missing in keyword
+}
+
+Test-Case: error
+Description: let-expr when variables are declared outside its' scope
+Labels: let-expr, int
+
+function errorFunction() {
+    int i = 2;
+    io:println(let int i = 2 in i); // @error redeclared symbol 'i'
+}
+
+Test-Case: error
+Description: let-expr when the type of let-var-decl does not match
+Labels: let-expr
+
+function errorFunction() {
+    io:println(let string i = "2", int j = 3 in i + j); // @error operator '+' not defined for 'string' and 'int'
+    io:println(let int i = 2 in foo(i)); // @error incompatible types: expected 'string?', found 'int'
+    io:println(let var i = 2 in foo(i)); // @error incompatible types: expected 'string?', found 'int'
+}
+
+function foo(string? s) returns string? {
+    return s;
+}
+
+Test-Case: output
+Description: Test let-expr as an isolated expression
```

Comment:
We need to include more details about the test.

e.g.,
```suggestion
Description: Test let-expr being an isolated expression, when the expression following `in` is isolated. Referenced variables bound by a let-var-decls are initialized using isolated expressions.
```

Reply:
We can also add a test when the expression following in is a different isolated expression (not referring to variables bound by let-var-decls).

Reply:
Added a test